test/file.jl

```diff
@@ -1431,4 +1431,20 @@ end
             @test readdir(b, join=true) ≛ [joinpath(b, x) for x in names]
         end
     end
+    if !Sys.iswindows()
+        mktempdir() do dir
+            cd(dir)
+            d = pwd() # might resolve symlinks
+            @test isdir(d)
+            @test Base.Filesystem.samefile(d, ".")
+            @test isempty(readdir())
+            @test isempty(readdir(d))
+            @test isempty(readdir(join=true))
+            rm(d, recursive=true)
+            @test !ispath(d)
+            @test isempty(readdir())
+            @test_throws SystemError readdir(d)
+            @test_throws Base.IOError readdir(join=true)
```

Comment:
It's kind of nasty that these throw different errors: the SystemError comes from trying to open a non-existend directory whereas the `Base.IOError` comes from trying to do `pwd()` in a deleted directory.